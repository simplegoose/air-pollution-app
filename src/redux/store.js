import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import cityReducer from './cities/citiesSlice';

const store = configureStore({
  reducer: {
    cities: cityReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
