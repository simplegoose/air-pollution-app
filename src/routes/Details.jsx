import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams, useSearchParams } from 'react-router-dom';
import hero from '../assets/img/air-pollution.png';
import DetailCard from '../components/DetailCard';
import { getPollution } from '../redux/cities/citiesSlice';

const Details = () => {
  const { name } = useParams();
  const [params] = useSearchParams();
  const lat = params.get('lat');
  const lon = params.get('lon');

  const dispatch = useDispatch();

  const { cities: { cities, isLoading, error } } = useSelector((state) => state);
  const city = cities.find((city) => city.city.toLowerCase() === name?.toLowerCase());

  useEffect(() => {
    dispatch(getPollution({ lat, lon, name }));
  }, [dispatch, lat, lon, name]);

  return (
    <main className="details">
      <section className="main-section">
        <div className="hero">
          <figure className="img-wrapper">
            <img src={hero} alt="Pollution" className="hero-img" />
          </figure>
          <div className="title">
            <h2>{city?.city}</h2>
          </div>
        </div>
        <h4 className="stats-header">CITY POLLUTION BREAKDOWN</h4>
        <div className="details-wrapper">
          {
            isLoading
            && <span className="loading">Loading...</span>
          }
          {
            error
            && <span className="error">An error has occured :-(</span>
          }
          {
            city?.list
            && city
              ?.list
              .map((listItem) => (
                <DetailCard
                  key={listItem.dt}
                  dt={listItem.dt}
                  components={listItem.components}
                />
              ))
        }
        </div>
      </section>
    </main>
  );
};

export default Details;
