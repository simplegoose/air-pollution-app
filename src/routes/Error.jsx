import React from 'react';

const Error = () => (
  <div className="error-page">
    <span>404 | Page not found</span>
  </div>
);

export default Error;
