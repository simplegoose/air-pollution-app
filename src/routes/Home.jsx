import React from 'react';
import { useSelector } from 'react-redux';
import hero from '../assets/img/air-pollution.png';
import CityCard from '../components/CityCard';

const Home = () => {
  const { cities: { cities } } = useSelector((state) => state);

  return (
    <main className="home">
      <section className="main-section">
        <div className="hero">
          <figure className="img-wrapper">
            <img src={hero} alt="Pollution" className="hero-img" />
          </figure>
          <div className="title">
            <h2>MAJOR CITIES</h2>
            <h3>Air Pollution</h3>
          </div>
        </div>
        <h4 className="stats-header">STATS BY CITY</h4>
        <div className="cities">
          {
            cities
            && cities.map((city) => <CityCard key={city.city} city={city} />)
          }
        </div>
      </section>
    </main>
  );
};

export default Home;
