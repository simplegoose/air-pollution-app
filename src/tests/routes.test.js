import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import Details from '../routes/Details';
import Home from '../routes/Home';
import Error from '../routes/Error';

describe('Test for routes', () => {
  it('Details page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Details />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Home page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Home />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Error page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Error />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
