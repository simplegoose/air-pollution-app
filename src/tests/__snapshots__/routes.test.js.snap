// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for routes Details page renders correctly 1`] = `
<main
  className="details"
>
  <section
    className="main-section"
  >
    <div
      className="hero"
    >
      <figure
        className="img-wrapper"
      >
        <img
          alt="Pollution"
          className="hero-img"
          src="air-pollution.png"
        />
      </figure>
      <div
        className="title"
      >
        <h2 />
      </div>
    </div>
    <h4
      className="stats-header"
    >
      CITY POLLUTION BREAKDOWN
    </h4>
    <div
      className="details-wrapper"
    >
      <span
        className="loading"
      >
        Loading...
      </span>
    </div>
  </section>
</main>
`;

exports[`Test for routes Error page renders correctly 1`] = `
<div
  className="error-page"
>
  <span>
    404 | Page not found
  </span>
</div>
`;

exports[`Test for routes Home page renders correctly 1`] = `
<main
  className="home"
>
  <section
    className="main-section"
  >
    <div
      className="hero"
    >
      <figure
        className="img-wrapper"
      >
        <img
          alt="Pollution"
          className="hero-img"
          src="air-pollution.png"
        />
      </figure>
      <div
        className="title"
      >
        <h2>
          MAJOR CITIES
        </h2>
        <h3>
          Air Pollution
        </h3>
      </div>
    </div>
    <h4
      className="stats-header"
    >
      STATS BY CITY
    </h4>
    <div
      className="cities"
    >
      <a
        className=""
        href="/details/London?lat=undefined&lon=undefined"
        onClick={[Function]}
      >
        <article
          className="city"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <div
            className="titles"
          >
            <h5>
              London
            </h5>
            <h6>
              Country:
               
            </h6>
          </div>
        </article>
      </a>
      <a
        className=""
        href="/details/Beijing?lat=undefined&lon=undefined"
        onClick={[Function]}
      >
        <article
          className="city"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <div
            className="titles"
          >
            <h5>
              Beijing
            </h5>
            <h6>
              Country:
               
            </h6>
          </div>
        </article>
      </a>
      <a
        className=""
        href="/details/Barcelona?lat=undefined&lon=undefined"
        onClick={[Function]}
      >
        <article
          className="city"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <div
            className="titles"
          >
            <h5>
              Barcelona
            </h5>
            <h6>
              Country:
               
            </h6>
          </div>
        </article>
      </a>
      <a
        className=""
        href="/details/Tokyo?lat=undefined&lon=undefined"
        onClick={[Function]}
      >
        <article
          className="city"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <div
            className="titles"
          >
            <h5>
              Tokyo
            </h5>
            <h6>
              Country:
               
            </h6>
          </div>
        </article>
      </a>
      <a
        className=""
        href="/details/Berlin?lat=undefined&lon=undefined"
        onClick={[Function]}
      >
        <article
          className="city"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <div
            className="titles"
          >
            <h5>
              Berlin
            </h5>
            <h6>
              Country:
               
            </h6>
          </div>
        </article>
      </a>
      <a
        className=""
        href="/details/Prague?lat=undefined&lon=undefined"
        onClick={[Function]}
      >
        <article
          className="city"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <div
            className="titles"
          >
            <h5>
              Prague
            </h5>
            <h6>
              Country:
               
            </h6>
          </div>
        </article>
      </a>
      <a
        className=""
        href="/details/Washington?lat=undefined&lon=undefined"
        onClick={[Function]}
      >
        <article
          className="city"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <div
            className="titles"
          >
            <h5>
              Washington
            </h5>
            <h6>
              Country:
               
            </h6>
          </div>
        </article>
      </a>
      <a
        className=""
        href="/details/Boston?lat=undefined&lon=undefined"
        onClick={[Function]}
      >
        <article
          className="city"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <div
            className="titles"
          >
            <h5>
              Boston
            </h5>
            <h6>
              Country:
               
            </h6>
          </div>
        </article>
      </a>
      <a
        className=""
        href="/details/Paris?lat=undefined&lon=undefined"
        onClick={[Function]}
      >
        <article
          className="city"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <div
            className="titles"
          >
            <h5>
              Paris
            </h5>
            <h6>
              Country:
               
            </h6>
          </div>
        </article>
      </a>
      <a
        className=""
        href="/details/Nairobi?lat=undefined&lon=undefined"
        onClick={[Function]}
      >
        <article
          className="city"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <div
            className="titles"
          >
            <h5>
              Nairobi
            </h5>
            <h6>
              Country:
               
            </h6>
          </div>
        </article>
      </a>
      <a
        className=""
        href="/details/Johannesburg?lat=undefined&lon=undefined"
        onClick={[Function]}
      >
        <article
          className="city"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <div
            className="titles"
          >
            <h5>
              Johannesburg
            </h5>
            <h6>
              Country:
               
            </h6>
          </div>
        </article>
      </a>
      <a
        className=""
        href="/details/Manchester?lat=undefined&lon=undefined"
        onClick={[Function]}
      >
        <article
          className="city"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <div
            className="titles"
          >
            <h5>
              Manchester
            </h5>
            <h6>
              Country:
               
            </h6>
          </div>
        </article>
      </a>
      <a
        className=""
        href="/details/Istanbul?lat=undefined&lon=undefined"
        onClick={[Function]}
      >
        <article
          className="city"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <div
            className="titles"
          >
            <h5>
              Istanbul
            </h5>
            <h6>
              Country:
               
            </h6>
          </div>
        </article>
      </a>
      <a
        className=""
        href="/details/Madrid?lat=undefined&lon=undefined"
        onClick={[Function]}
      >
        <article
          className="city"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <div
            className="titles"
          >
            <h5>
              Madrid
            </h5>
            <h6>
              Country:
               
            </h6>
          </div>
        </article>
      </a>
      <a
        className=""
        href="/details/Munich?lat=undefined&lon=undefined"
        onClick={[Function]}
      >
        <article
          className="city"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <div
            className="titles"
          >
            <h5>
              Munich
            </h5>
            <h6>
              Country:
               
            </h6>
          </div>
        </article>
      </a>
      <a
        className=""
        href="/details/Rome?lat=undefined&lon=undefined"
        onClick={[Function]}
      >
        <article
          className="city"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <div
            className="titles"
          >
            <h5>
              Rome
            </h5>
            <h6>
              Country:
               
            </h6>
          </div>
        </article>
      </a>
      <a
        className=""
        href="/details/Athens?lat=undefined&lon=undefined"
        onClick={[Function]}
      >
        <article
          className="city"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <div
            className="titles"
          >
            <h5>
              Athens
            </h5>
            <h6>
              Country:
               
            </h6>
          </div>
        </article>
      </a>
      <a
        className=""
        href="/details/Brussels?lat=undefined&lon=undefined"
        onClick={[Function]}
      >
        <article
          className="city"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <div
            className="titles"
          >
            <h5>
              Brussels
            </h5>
            <h6>
              Country:
               
            </h6>
          </div>
        </article>
      </a>
      <a
        className=""
        href="/details/Kyiv?lat=undefined&lon=undefined"
        onClick={[Function]}
      >
        <article
          className="city"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowCircleRightOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
            />
          </svg>
          <div
            className="titles"
          >
            <h5>
              Kyiv
            </h5>
            <h6>
              Country:
               
            </h6>
          </div>
        </article>
      </a>
    </div>
  </section>
</main>
`;
